### ISSUE TITLE
(Costume-group) Animation issues

### ISSUE THREAD

**Issue by nighca** [goplus/spx#300]
Project: [animation.zip](https://github.com/user-attachments/files/16094508/animation.zip)

For sprite `Fighter`:

Config:

```json
{
  "fAnimations": {
    "fight": {
      "from": "__animation_fight_attack_1-1",
      "to": "__animation_fight_attack_1-4",
      "duration": 0.4,
      "anitype": 0
    },
    "dying": {
      "from": "__animation_dying_dead-1",
      "to": "__animation_dying_dead-3",
      "duration": 0.6,
      "anitype": 0
    },
    "walk": {
      "from": "__animation_walk_walk-1",
      "to": "__animation_walk_walk-8",
      "duration": 0.8,
      "anitype": 0
    },
    "default": {
      "from": "__animation_default_idle-1",
      "to": "__animation_default_idle-6",
      "duration": 0.6,
      "anitype": 0
    }
  },
  "defaultAnimation": "default",
  "animBindings": {
    "die": "dying",
    "step": "walk"
  }
}
```

Code:

```go
onKey KeyUp, => {
    say "animate fight", 0.3
    animate "fight"
}

onKey KeyDown, => {
    say "die", 0.3
    die
}

onKey KeyLeft, => {
    say "step left"
    setRotationStyle LeftRight
    turnTo Left
    step 50
}

onKey KeyRight, => {
    say "step right"
    setRotationStyle LeftRight
    turnTo Right
    step 50
}
```

There are some issues:

1. The animation `default` (which is bound as `defaultAnimation`) is played only once while sprite in default state. It is expecetd to be played repeatedly (see details in https://github.com/goplus/builder/issues/603#issuecomment-2180367864)
2. After animation `fight` (played by calling `animate "fight"`) finished playing, the sprite does not return to the default state, while it keeps the last csotume of animation `fight`
3. If we call `step 50`, it is expected to play animation `walk` (which is bound to state `step`). However, the sprite plays `fight`, and then `die`, and then `walk`, and then panic:

    ```
    panic: invalid costume index [recovered]
        panic: invalid costume index
    ```

**Comment by nighca** [comment]
4. To make it possible to bind one animation to multiple states at the same time, we should put all animations in one place, e.g., `fAnimations`.

**Comment by nighca** [comment]
Now in builder, all the animations are frame animations, based on https://github.com/goplus/spx/pull/303 we do configure like this:

* Set `frameFrom` & `frameTo` to specify frames, do not set `from` & `to`
* Set `duration` to a fixed value (whatever the project or sprite or animation is), which corresponds to the time cost for sprite to move distance `1`
* Set `frameFps` to control the speed of frame animation, do not set `fps`
* Do not set `anitype`

**Comment by nighca** [comment]
fixed by #303
